blockchain/blockchain.go

```diff
@@ -123,18 +123,10 @@ func (b *Blockchain) Head() (*core.Block, error) {
 }
 
 func (b *Blockchain) HeadsHeader() (*core.Header, error) {
-	var header *core.Header
-	return header, b.database.View(func(txn db.Transaction) error {
-		height, err := chainHeight(txn)
-		if err != nil {
-			return err
-		}
-
-		header, err = blockHeaderByNumber(txn, height)
-		return err
-	})
+	return headsHeader(b.database)
```

Comment:
see https://github.com/NethermindEth/juno/pull/857/files#diff-37b6e7ed4f299b87b357b54376123393b34ee0699fc035f25bc0621627dcc6baL116-L124

you need to use View function on the database

Reply:
okay I added the view function on the database. hope it's correct.

also I'm really sorry for making any mistakes since I'm just a newbie to all this.

Reply:
No worries!